### PR TITLE
work in Backward Euler

### PR DESCRIPTION
**Pull request by jorgepz**

### CODE REVIEW

File: src/Algorithms/BackwardEuler.jl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+BackwardEuler(; Δt::T) where N = BackwardEuler(Δt)
+
+function _init(alg::BackwardEuler, M, C, K)
+    Δt = alg.Δt
+    K̂ = K * Δt + C
+    return K̂
+end
+
+function _solve(alg::BackwardEuler{N},
+                ivp::InitialValueProblem{<:SecondOrderAffineContinuousSystem{N}, XT},
```
Comment (mforets) [inline src/Algorithms/BackwardEuler.jl:24]: here we could dispatch on `Ex'(t) = Ax(t) + f(t)`, which is the purpose of https://github.com/JuliaReach/MathematicalSystems.jl/issues/230

File: src/Algorithms/BackwardEuler.jl [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                NSTEPS::Int) where {N, VT, XT<:Tuple{VT, VT}}
+
+    sys = system(ivp)
+    (U₀) = initial_state(ivp)
```
Comment (mforets) [inline src/Algorithms/BackwardEuler.jl:28]:
```suggestion
    (U₀, _) = initial_state(ivp)
```

File: src/Algorithms/BackwardEuler.jl (hunk 1 above)
Comment (mforets) [reply src/Algorithms/BackwardEuler.jl:24]: ... but we can do it in a later step, in the meantime assuming `M  =  0`

File: src/Algorithms/BackwardEuler.jl [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    IMAX = NSTEPS + 1
+    M, C, K, R = _unwrap(sys, IMAX)
```
Comment (mforets) [inline src/Algorithms/BackwardEuler.jl:31]: after this line we can add this check

````julia
iszero(M) || throw(ArgumentError("this method assumes that the system is of first order"))
```

File: src/Algorithms/BackwardEuler.jl [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+    U₀′′ = M \ (R[1] - C * U₀′ - K * U₀)
```
Comment (mforets) [inline src/Algorithms/BackwardEuler.jl:33]:
```suggestion
```

File: src/Algorithms/BackwardEuler.jl (hunk 1 above)
Comment (jorgepz) [reply src/Algorithms/BackwardEuler.jl:24]: ok

File: src/Algorithms/BackwardEuler.jl (hunk 3 above)
Comment (jorgepz) [reply src/Algorithms/BackwardEuler.jl:33]: done in commit https://github.com/ONSAS/StructuralDynamicsODESolvers.jl/pull/7/commits/318910365815ec1271979431c907bcafd2cf0810

**Review by mforets — APPROVED** [review]
vamo :uruguay:

**Comment by jorgepz** [comment]
closes #6

**Comment by mforets** [comment]
i'll merge when tests pass..

**Pull request merged**